File: TransCoder_tokenized_test_set_functions/cpp/COUNT_STRINGS_ADJACENT_CHARACTERS_DIFFERENCE_ONE.cpp
int countStrs ( int n ) {
  long int dp [ n + 1 ] [ 27 ];
  memset ( dp, 0, sizeof ( dp ) );
  for ( int i = 0;
  i <= 25;
  i ++ ) dp [ 1 ] [ i ] = 1;
  for ( int i = 2;
  i <= n;
  i ++ ) {
    for ( int j = 0;
    j <= 25;
    j ++ ) if ( j == 0 ) dp [ i ] [ j ] = dp [ i - 1 ] [ j + 1 ];
    else dp [ i ] [ j ] = ( dp [ i - 1 ] [ j - 1 ] + dp [ i - 1 ] [ j + 1 ] );
  }
  long int sum = 0;
  for ( int i = 0;
  i <= 25;
  i ++ ) sum = ( sum + dp [ n ] [ i ] );
  return sum;
}

File: results/java_codex/COUNT_STRINGS_ADJACENT_CHARACTERS_DIFFERENCE_ONE.java
long countStrs(int n) {
  long dp[n + 1][27];
  for (int i = 0; i < n + 1; i++) {
    for (int j = 0; j < 27; j++) {
      dp[i][j] = 0;
    }
  }
  for (int i = 0; i <= 25; i++) {
    dp[1][i] = 1;
  }
  for (int i = 2; i <= n; i++) {
    for (int j = 0; j <= 25; j++) {
      if (j == 0) {
        dp[i][j] = dp[i - 1][j + 1];
      } else {
        dp[i][j] = (dp[i - 1][j - 1] + dp[i - 1][j + 1]);
      }
    }
  }
  long sum = 0;
  for (int i = 0; i <= 25; i++) {
    sum = (sum + dp[n][i]);
  }
  return sum;
}